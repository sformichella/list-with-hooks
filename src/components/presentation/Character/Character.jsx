import React from 'react';
import PropTypes from 'prop-types';

export default function Character({ id, name, image }) {
  return (
    <>
      <a href={`/${id}`}>{name}</a>
      <img src={image}/>
    </>
  );
}

Character.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};
