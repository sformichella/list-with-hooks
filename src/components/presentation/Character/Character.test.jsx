import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Character from './Character';

describe('Character component', () => {
  afterEach(() => cleanup());
  it('renders Character', () => {
    const { asFragment } = render(<Character 
      id={1}
      name="Cool name"
      image="character-image.com"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
