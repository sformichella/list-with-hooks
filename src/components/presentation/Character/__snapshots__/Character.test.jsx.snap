// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <a
    href="/1"
  >
    Cool name
  </a>
  <img
    src="character-image.com"
  />
</DocumentFragment>
`;
